packages/FormElement/src/FormElement.js

```diff
@@ -66,117 +70,91 @@ function FormElement(props) {
         id: ariaDescriptionId,
       });
     }
 
     return null;
   }
 
-  function renderInstructions() {
-    if (extractedChildren["FormElement.Instructions"]) {
-      return React.cloneElement(extractedChildren["FormElement.Instructions"], {
-        id: ariaInstructionsId,
-      });
-    }
-    return null;
+  function renderContent() {
+    return getClonedElement(subComponentDisplayNames.Content, {
+      idForLabel,
+      refLabel,
+      ariaDescribedBy: `${ariaErrorId} ${ariaInstructionsId} ${ariaDescriptionId}`,
+    });
   }
 
-  const nativeChildTypes = ["input", "textarea", "select"];
-
-  const renderFormElementChild = child => (
-    <div data-pka-anchor="form-element.child" key={child.key} css={formElementChildStyle}>
-      {child}
-    </div>
-  );
-
-  const isNativeElement = child => isString(child.type) && nativeChildTypes.includes(child.type);
-
-  const childExtendedProps = {
-    "aria-describedby": `${ariaErrorId} ${ariaInstructionsId} ${ariaDescriptionId}`,
-    id: inputId,
-    "aria-required": hasRequiredLabel,
-    "aria-invalid": hasError,
-  };
-
+  function renderInstructions() {
+    return getClonedElement(subComponentDisplayNames.Instructions, {
+      id: ariaInstructionsId,
+    });
+  }
   return (
     <div css={formElementStyles} isInline={isInline} size={size} isDisabled={isDisabled} {...moreProps}>
       <Label
         hasOptionalLabel={hasOptionalLabel}
         hasRequiredLabel={hasRequiredLabel}
-        help={extractedChildren["FormElement.Help"]}
-        id={inputId}
+        help={extractedChildren[subComponentDisplayNames.Help]}
+        id={idForLabel}
         isInline={isInline}
         isVisuallyHidden={isLabelVisuallyHidden}
         label={label}
         ref={refLabel}
       />
       <div css={isInline ? inlineContainerStyles : null}>
-        {renderInstructions()}
-        {extractedChildren.children.map(child => {
-          const extendedProps = isNativeElement(child)
-            ? {
-                ...childExtendedProps,
-                disabled: isDisabled,
-                readOnly: isReadOnly,
-              }
-            : {
-                ...childExtendedProps,
-                refLabel,
-                hasError,
-                isDisabled,
-                isReadOnly,
-                size,
-                "aria-disabled": isDisabled,
-                "aria-readonly": isReadOnly,
-              };
-
-          return renderFormElementChild(React.cloneElement(child, extendedProps));
-        })}
+        {extractedChildren[subComponentDisplayNames.Instructions] ? renderInstructions() : null}
+        {extractedChildren[subComponentDisplayNames.Content] ? renderContent() : null}
         {renderFooter()}
       </div>
     </div>
   );
 }
 
 FormElement.displayName = "FormElement";
-
 FormElement.propTypes = propTypes;
 FormElement.defaultProps = defaultProps;
-
+FormElement.SubComponentDisplayNames = subComponentDisplayNames;
```

Comment:
```suggestion
FormElement.subComponentDisplayNames = subComponentDisplayNames;
```

Reply:
So that this doesn't seem like an actual component. It also aligns with the decision recently made here:

https://github.com/acl-services/paprika/pull/392